packages/loopring_v3/contracts/impl/LoopringV3.sol

```diff
@@ -251,20 +227,22 @@ contract LoopringV3 is ILoopringV3, Claimable
         uint amountLRC
         )
         external
+        nonReentrant
         returns (uint stakedLRC)
     {
         require(amountLRC > 0, "ZERO_VALUE");
         require(
-            lrcAddress.safeTransferFrom(
-                msg.sender,
-                address(this),
-                amountLRC
-            ),
+            lrcAddress.safeTransferFrom(msg.sender, address(this), amountLRC),
             "TRANSFER_FAILURE"
         );
-        stakedLRC = exchanges[exchangeId - 1].exchangeStake.add(amountLRC);
-        exchanges[exchangeId - 1].exchangeStake = stakedLRC;
+
+        Exchange storage exchange = exchanges[exchangeId];
+        require(exchange.exchangeAddress != address(0), "INVALID_EXCHANGE_ID");
```

Comment:
It's better to add exchangeID check before transferring real token.

Reply:
this contract allows `exchangeId` to be 0, `ProrotocolRegister` may have stricter rules. Therefore we only care if the id's corresponding address is address(0) or not.

Reply:
What I mean is not exchangeID check 0 or not. I mean we should first require exchangeAddress cannot be zero address ahead of calling safeTransferFrom.

We need switch the two require statement's sequence just like in withdrawExchangeStake.